lib/engine/game/g_1848/game.rb

```diff
@@ -658,6 +688,79 @@ def market_share_limit(corporation = nil)
 
           MARKET_SHARE_LIMIT
         end
+
+        def can_take_loan?(entity)
+          entity.corporation? &&
+            entity.loans.size < maximum_loans(entity) &&
+            @loans.any?
+        end
+
+        def take_loan(entity, loan)
+          raise GameError, "Cannot take more than #{maximum_loans(entity)} loans" unless can_take_loan?(entity)
+
+          old_price = entity.share_price
+          boe_old_price = @boe.share_price
+          name = entity.name
+          @log << "#{name} takes a loan and receives #{format_currency(loan.amount)}"
+          @boe.spend(loan.amount, entity)
+          loan_taken_stock_market_movement(entity)
```

Comment:
I'd personally put all the share price movement logging inside this method instead of here. Up to you.